astropy/units/decorators.py

```diff
@@ -222,7 +222,7 @@ def wrapper(*func_args, **func_kwargs):
                 #    are not strings or subclasses of Unit. This is to allow
                 #    non unit related annotations to pass through
                 if is_annotation:
-                    valid_targets = [t for t in valid_targets if isinstance(t, (str, Unit))]
+                    valid_targets = [t for t in valid_targets if isinstance(t, (str, UnitBase))]
```

Comment:
I appreciate I never got to reviewing #8984 but it makes me sad that `u.quantity_input` has evolved to only work with astropy units. In it's first incarnation it was type agnostic.

Reply:
Also it should be noted that I am pretty sure that we lost the duck typing in `quantity_input` long ago ( so this comment was really pointless)

Reply:
Well, well, you should have written a test for it if you wanted it to be supported... 😝